### ISSUE TITLE
python3.5.[01]: Union[Pattern, str] raises an error

### ISSUE THREAD

**Issue by asottile** [asottile/flake8-typing-imports#6]
Need to make sure this gets quoted instead of left unquoted:

```pycon
>>> Union[AnyStr, str]
typing.Union[~AnyStr, str]
>>> Union[str, Pattern]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.5/typing.py", line 534, in __getitem__
    dict(self.__dict__), parameters, _root=True)
  File "/usr/local/lib/python3.5/typing.py", line 491, in __new__
    for t2 in all_params - {t1} if not isinstance(t2, TypeVar)):
  File "/usr/local/lib/python3.5/typing.py", line 491, in <genexpr>
    for t2 in all_params - {t1} if not isinstance(t2, TypeVar)):
TypeError: issubclass() arg 1 must be a class
```

**Comment by asottile** [comment]
The hint is to add a `visit_Subscript` to here: https://github.com/asottile/flake8-typing-imports/blob/292b2d518997efab0e2b1963a39f0d9ded9b39a3/flake8_typing_imports.py#L440

And then report the error in the flake8 plugin below

Probably need to check for `Pattern` exactly, there may be other types that have problems as well

The nice thing about suggesting quoting is it won't show up in the ast because they're just strings!

**Comment by atugushev** [comment]
Seems like only affects `Pattern` and `Match`.

> Probably need to check for `Pattern` exactly, there may be other types that have problems as well

Should `typing.Pattern` (as `ast.Attribute`, not `ast.Name`) be checked as well? For cases like:
```python
def foo(bar: typings.Union[typings.Pattern, str])
```

**Comment by asottile** [comment]
yeah probably handle both `typing.Pattern` and `Pattern` -- the visitor already records the `from` import.  if you're looking for prior art, `pyupgrade` does similar things for `six` imports

**Comment by atugushev** [comment]
Thanks for the hint!

**Comment by atugushev** [comment]
@asottile 

Could you help me with an error text? I came up with:

1. TYP003 Union[Pattern, ...] or Union[Match, ...] is broken in <3.5.2
1. TYP003 Union[Pattern, Match, ...] is broken in <3.5.2
1. TYP003 Union[Pattern, ...] or Union[Match, ...] is broken in <3.5.2, consider quoting type annotation
1. TYP003 Union[Pattern, Match, ...] is broken in <3.5.2, consider quoting type annotation

**Comment by asottile** [comment]
available in 1.5.0!
